cmd/consensus/main.go

```diff
@@ -238,92 +239,68 @@ func main() {
 			return err
 		}).
 		Module("random beacon key", func(node *cmd.NodeConfig) error {
-			// If this node was a participant in a spork, their DKG key for the
+			// If this node was a participant in a spork, their beacon key for the
 			// first epoch was generated during the bootstrapping process and is
 			// specified in a private bootstrapping file. We load their key and
 			// store it in the db for the initial post-spork epoch for use going
 			// forward.
+			//
 			// If this node was not a participant in a spork, they joined at an
-			// epoch boundary, so they have no DKG file (they will generate
-			// their first DKG private key through the procedure run during the
-			// current epoch setup phase), and we do not need to insert a key at
-			// startup.
+			// epoch boundary, so they have no beacon key file (they will generate
+			// their first beacon private key through the DKG in the EpochSetup phase
+			// prior to their first epoch as network participant).
 
-			// if the node is not part of the current epoch identities, we do
-			// not need to load the key
-			rootEpoch := node.State.AtBlockID(node.RootBlock.ID()).Epochs().Current()
-			initialIdentities, err := rootEpoch.InitialIdentities()
+			rootSnapshot := node.State.AtBlockID(node.RootBlock.ID())
+			isSporkRoot, err := protocol.IsSporkRootSnapshot(rootSnapshot)
 			if err != nil {
-				return err
+				return fmt.Errorf("could not check whether root snapshot is spork root: %w", err)
 			}
-			if _, ok := initialIdentities.ByNodeID(node.NodeID); !ok {
-				node.Logger.Info().Msg("node joined at epoch boundary, not reading DKG file")
+			if !isSporkRoot {
+				node.Logger.Info().Msg("node starting from mid-spork snapshot, will not read spork random beacon key file")
 				return nil
 			}
 
-			// otherwise, load and save the key in DB for the current epoch (wrt
-			// root block)
+			// If the node has a beacon key file, then save it to the secrets database
+			// as the beacon key for the first epoch of the root snapshot.
```

Comment:
```suggestion
			// as the beacon key for the epoch of the root snapshot.
```

I found the `first` here is a bit confusing, since a snapshot can only belong to one epoch.